tokio/src/macros/select.rs

```diff
@@ -0,0 +1,788 @@
+/// Wait on multiple concurrent branches, returning when the **first** branch
+/// completes, cancelling the remaining branches.
+///
+/// The `select` macro accepts one or more branches with the following pattern:
+///
+/// ```text
+/// <pattern> = <async expression> (, if <condition>)? => <handler>,
```

Comment:
I have no strong opinion on that, but I guess starting with the condition could also be an option to signal that this one gets evaluated first:

```
(if <condition>:)? <pattern> = <async expression> => <handler>,
```

Reply:
Unfortunately, I don't think this would work. `condition` needs to be `$condition:expr`, which means it must be followed by `=>`, `,`, or `;`.